server/src/main/java/org/apache/druid/client/DirectDruidClient.java

```diff
@@ -207,9 +207,12 @@ public int getNumOpenConnections()
         private final AtomicBoolean done = new AtomicBoolean(false);
         private final AtomicReference<String> fail = new AtomicReference<>();
         private final AtomicReference<TrafficCop> trafficCopRef = new AtomicReference<>();
+        private final AtomicReference<Boolean> isChannelReadable = new AtomicReference<>();
```

Comment:
I would use `AtomicBoolean` not `AtomicReference<Boolean>`. And I think change variable name to `channelReadable` would be better since there are another variable `private final AtomicBoolean done = new AtomicBoolean(false)` in this file, it's not start with `is`.

Reply:
Thanks for the review @QiuMM, really appreciate it. 

I have fixed the test that was causing these errors. Could you confirm if the logic behind the metric makes sense please?